### DISCUSSION TITLE
What is the right way to access prompt/model generation (was .text in 0.0.63)?

### DISCUSSION THREAD

**Discussion by dchichkov** [guidance-ai/guidance#616] [Q&A] [answered]
Hi. A question. I've migrated my code to 0.1 from 0.0.63.  And .text variable that used to contain complete prompt is no longer available :-/ .  

What is the right way to access completed prompt / generation?  
Can .text field be brought back?
Or even better, for "with system() ..." style, produce a typical {role: "system", content: "..."} array?

(I'm validating these and then using validated prompts/generations to finetune the LLM.)

**Comment by Harsha-Nori** [comment, marked as answer]
@dchichkov , you can access the fully built up state of any LM object just by calling `str(lm)`. For example:

<img width="1114" alt="image" src="https://github.com/guidance-ai/guidance/assets/5302119/52e4900f-50b9-4714-a1c0-019f64d91c09">

Is this what you're looking for? The output of `str(lm)` is just a standard Python string, so you can manipulate it however you like.

**Reply by dchichkov** [reply]
Thanks.  It'be great also, if for chat models a conv [{"role":...}, {} ..   was available.

**Reply by Harsha-Nori** [reply]
Good idea -- might not be too hard to build a simple extractor for this. I assume you're just expecting something compliant with the standard OpenAI ChatCompletions `messages` field right?
